packages/flutter_test/lib/src/binding.dart

```diff
@@ -849,20 +849,15 @@ abstract class TestWidgetsFlutterBinding extends BindingBase
   }
 
   Future<void> _handleAnnouncementMessage(Object? mockMessage) async {
-    final Map<Object?, Object?> message = mockMessage! as Map<Object?, Object?>;
-    if (message['type'] == 'announce') {
-      final Map<Object?, Object?> data =
-          message['data']! as Map<Object?, Object?>;
-      final String dataMessage = data['message'].toString();
-      final TextDirection textDirection =
-          TextDirection.values[data['textDirection']! as int];
-      final int assertivenessLevel = (data['assertiveness'] as int?) ?? 0;
-      final Assertiveness assertiveness =
-          Assertiveness.values[assertivenessLevel];
-      final CapturedAccessibilityAnnouncement announcement =
-          CapturedAccessibilityAnnouncement._(
-              dataMessage, textDirection, assertiveness);
-      _announcements.add(announcement);
+    if (mockMessage! case {
+      'type': 'announce',
+      'data': final Map<Object?, Object?> data as Map<Object?, Object?>,
```

Comment:
Rather than silently returning when `data` is in an invalid format, the [cast pattern](https://dart.dev/language/pattern-types#cast) will throw an error, similar to
`final Map<Object?, Object?> data =
          message['data']! as Map<Object?, Object?>;`

<br>

Ideally, it would just be 

```dart
    if (mockMessage! case {
      'type': 'announce',
      'data': data as Map<Object?, Object?>,
    }) {
      // ...
    }
```

but that sadly doesn't play nice with `always_specify_types`.

Reply:
Makes sense!